include/cuco/detail/utils.hpp

```diff
@@ -49,5 +51,12 @@ auto get_grid_size(Kernel kernel, std::size_t block_size, std::size_t dynamic_sm
   return grid_size;
 }
 
+template <typename Iterator>
+constexpr inline int64_t distance(Iterator begin, Iterator end)
+{
+  // `int64_t` instead of arch-dependant `long int`
+  return static_cast<int64_t>(std::distance(begin, end));
```

Comment:
Do we also need a `thrust::distance` version is it 100% equivalent to the `std::` implementation? I'm a little afraid that there might be some weird thrust iterator constructs that only work with `thrust::distance`.

Reply:
I believe that `thrust::` is needed only if called in device code. Otherwise `std::` would be enough.

Reply:
Ah, I see. And since we now have `cuda::std::distance` (I guess that's a thing) for both host and device code the thrust version should be obsolete.

Reply:
Right, this function is only invoked on the host side thus `std::distance` should be sufficient.